dali/kernels/slice/slice_flip_normalize_permute_pad_gpu.h

```diff
@@ -135,42 +238,25 @@ class SliceFlipNormalizePermutePadGpu {
       }
     }
 
-    detail::SampleDesc<Dims> *sample_descs =
-      context.scratchpad->Allocate<detail::SampleDesc<Dims>>(
-        AllocType::GPU, num_samples);
-
-    float *norm_add = mean_data.empty() ? nullptr :
-      context.scratchpad->Allocate<float>(
-        AllocType::GPU, mean_data.size());
-
-    float *norm_mul = inv_stddev_data.empty() ? nullptr :
-      context.scratchpad->Allocate<float>(
-        AllocType::GPU, inv_stddev_data.size());
-
-    detail::BlockDesc *block_descs =
-      context.scratchpad->Allocate<detail::BlockDesc>(
-        AllocType::GPU, block_count_);
-
-    // Memory is allocated contiguously, so we launch only one cudaMemcpyAsync
-    size_t total_bytes = num_samples * sizeof(detail::SampleDesc<Dims>)
-      + mean_data.size() * sizeof(float)
-      + inv_stddev_data.size() * sizeof(float)
-      + block_count_ * sizeof(detail::BlockDesc);
-    cudaMemcpyAsync(sample_descs, sample_descs_cpu,
-                    total_bytes,
-                    cudaMemcpyHostToDevice,
-                    context.gpu.stream);
-
-    const auto grid = block_count_;
-    if (norm_add != nullptr && norm_mul != nullptr) {
-      detail::SliceFlipNormalizePermutePadKernel<OutputType, InputType, Dims, true>
-          <<<grid, kBlockDim, 0, context.gpu.stream>>>(sample_descs, block_descs, norm_add,
-                                                       norm_mul, normalization_dim);
-    } else {
-      detail::SliceFlipNormalizePermutePadKernel<OutputType, InputType, Dims, false>
-          <<<grid, kBlockDim, 0, context.gpu.stream>>>(sample_descs, block_descs, norm_add,
-                                                       norm_mul, normalization_dim);
-    }
+    detail::SampleDesc<Dims> *sample_descs_gpu = nullptr;
+    detail::BlockDesc *block_descs_gpu = nullptr;
+    std::tie(sample_descs_gpu, block_descs_gpu) = context.scratchpad->ToContiguousGPU(
+        context.gpu.stream,
+        make_span(sample_descs_cpu, num_samples),
+        make_span(block_descs_cpu, block_count_));
+
+    VALUE_SWITCH(need_pad ? 1 : 0, NeedPad, (false, true), (
+      VALUE_SWITCH(need_flip ? 1 : 0, NeedFlip, (false, true), (
```

Comment:
Nitpick:  you can put a comment about this conversion to int like:
```// Convert switch argument to `int` to avoid compiler warning about unreachable case label```
or similar - otherwise someone might try to "fix" it.